File: apps/web/src/api/skills.js
import { getJson, patchJson, postJson } from "./client";

export const getSkills = (skillType = null, activeOnly = null) => {
  const params = new URLSearchParams();
  if (skillType) params.set("skill_type", skillType);
  if (activeOnly !== null) params.set("active_only", String(activeOnly));
  const query = params.toString();
  return getJson(`/api/skills${query ? `?${query}` : ""}`);
};

export const getSkill = (skillId) =>
  getJson(`/api/skills/${skillId}`);

export const createSkill = (body) =>
  postJson("/api/skills", body);

export const updateSkill = (skillId, body) =>
  patchJson(`/api/skills/${skillId}`, body);

File: apps/api/app/api/routes/skills.py
from fastapi import APIRouter, Depends, Query

from app.api.deps import get_pipeline_service
from app.schemas.skills import (
    SkillCreateRequest,
    SkillListResponse,
    SkillResponse,
    SkillUpdateRequest,
)
from app.services.pipeline_service import PipelineService

router = APIRouter()


@router.post("", response_model=SkillResponse)
async def create_skill(
    payload: SkillCreateRequest,
    service: PipelineService = Depends(get_pipeline_service),
) -> SkillResponse:
    return service.create_skill(payload)


@router.get("", response_model=SkillListResponse)
async def list_skills(
    skill_type: str | None = Query(default=None),
    active_only: bool | None = Query(default=None),
    service: PipelineService = Depends(get_pipeline_service),
) -> SkillListResponse:
    return service.list_skills(skill_type, active_only)


@router.get("/{skill_id}", response_model=SkillResponse)
async def get_skill(
    skill_id: str,
    service: PipelineService = Depends(get_pipeline_service),
) -> SkillResponse:
    return service.get_skill(skill_id)


@router.patch("/{skill_id}", response_model=SkillResponse)
async def update_skill(
    skill_id: str,
    payload: SkillUpdateRequest,
    service: PipelineService = Depends(get_pipeline_service),
) -> SkillResponse:
    return service.update_skill(skill_id, payload)
